identity-account/src/events/macros.rs

```diff
@@ -33,46 +33,47 @@ macro_rules! impl_command_builder {
   ($ident:ident { $(@ $requirement:ident $field:ident $ty:ty $(= $value:expr)?),* $(,)* }) => {
     paste::paste! {
       #[derive(Clone, Debug)]
-      pub struct [<$ident Builder>] {
+      pub struct [<$ident Builder>]<'account, K: $crate::identity::IdentityKey> {
+        account: &'account Account,
+        key: K,
         $(
           $field: Option<$ty>,
         )*
       }
 
-      impl [<$ident Builder>] {
+      impl<'account, K: $crate::identity::IdentityKey> [<$ident Builder>]<'account, K> {
         $(
           pub fn $field<VALUE: Into<$ty>>(mut self, value: VALUE) -> Self {
             self.$field = Some(value.into());
             self
           }
         )*
 
-        pub fn new() -> [<$ident Builder>] {
+        pub fn new(account: &'account Account, key: K) -> [<$ident Builder>]<'account, K> {
           [<$ident Builder>] {
+            account,
+            key,
             $(
               $field: None,
             )*
           }
         }
 
-        pub fn finish(self) -> $crate::Result<$crate::events::Command> {
-          Ok($crate::events::Command::$ident {
+        pub async fn apply(self) -> $crate::Result<()> {
+          let account = self.account;
+          let update = $crate::events::Command::$ident {
             $(
               $field: impl_command_builder!(@finish self $requirement $field $ty $(= $value)?),
             )*
-          })
+          };
+          account.apply_command(self.key, update).await?;
+          Ok(())
         }
```

Comment:
Did we decide for or against chaining consecutive updates? We could return a new `IdentityUpdater` on `apply()` to enable chaining, but I think it was decided not to be particularly useful.

Reply:
I'm not sure we reached a proper consensus on it, but in my opinion running one update is complex enough, that I wouldn't personally do chaining. I think with what is shown in the examples, we have a nice middle ground between the previous "two statements for one update" and "we could run multiple updates in one statement".